hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HiveHoodieReaderContext.java

```diff
@@ -229,13 +229,17 @@ public long getPos() throws IOException {
     if (firstRecordReader != null) {
       return firstRecordReader.getPos();
     }
-    throw new IllegalStateException("getPos() should not be called before a record reader has been initialized");
```

Comment:
Should we still throw `IllegalStateException` if the input split has a base file and `firstRecordReader` is `null`? Is that possible?

Reply:
Oops, I have landed it, we can address it in following up PRs if necessary.